### ISSUE TITLE
[Feature] Face Recognition

### ISSUE THREAD

**Issue by mubashir-rehman** [leonlatsch/Photok#124]
**Is your feature request related to a problem? Please describe.**
It is hard to find pictures of a person when scrolling through hundreds of pictures while most of which are memes. 

**Describe the solution you'd like**
A face recognition feature and then tagging images with faces and names assigned to those faces would be great. Along with that, a separate tab for list of detected faces would be marvelous. 

**Describe alternatives you've considered**
Gallery Go(a gallery app by Google) does a pretty good job at it.

**Additional context**
Picasa3D on Windows was an amazing piece of software which I use till date despite being deprecated by Google. It does all the  processing locally. I wanted same features of tagging and removing tags of faces to fix false positives manually. I couldn't find a good app for that in Android market that worked offline. Then GalleryGo came to rescue. But that app despite being minimal and doing the work. Doesn't give users ability to add or remove faces.

**Comment by comparesheriff** [comment]
This probably only can be implemented after #41 is done. Then Photok will have a base structure to build upon for folders/tags.

Additionally this should be deactivated by default, as I believe this isn't a priority for most users and will probably have an impact on the performance of the app while importing images (scanning every imported image for faces).

@leonlatsch I couldn't find the source of the Gallery Go app mentioned, but maybe in this [Medium Article about Facial Recognition in Android](https://medium.com/devnibbles/facial-recognition-with-android-1-4-5e043c264edc) or with [OpenCV for Android](https://opencv.org/android/) there are some ideas on how to implement this locally.

**Comment by mubashir-rehman** [comment]
GalleryGo is closed source. https://play.google.com/store/apps/details?id=com.google.android.apps.photosgo&hl=en&gl=US

**Comment by leonlatsch** [comment]
When you have a look at #41, the planned UI design for collections is already inspired by Gallery Go.

This feature would be an addition to the collections feature.

**Comment by mubashir-rehman** [comment]
I did look at the issue and was amazed that I think like others too. I have been using GalleryGo for its design and features for a long time. But it lacks ability to create a pin protected safe and adding tags to the pics. It does all the work by itself with no room for customization.

**Comment by leonlatsch** [comment]
This is kind of out of scope for Photok.
Feel free to fork 🙂
